### PR TITLE
Swap 1479 fixing proposal workflow bugs

### PR DESCRIPTION
**Pull request by martin-trajanovski**
## Description

Fixed couple of bugs when removing and inserting connections in the workflows. Also fixed some small workflow engine issues.

## How Has This Been Tested

e2e tests
manual tests

## Fixes

https://jira.esss.lu.se/browse/SWAP-1479

## Tests included/Docs Updated?

- [x] I have added tests to cover my changes.
- [ ] All relevant doc has been updated

### CODE REVIEW

File: src/datasources/mockups/ProposalSettingsDataSource.ts [hunk 1]
```diff
@@ -146,7 +146,14 @@ export class ProposalSettingsDataSourceMock
 
   async getProposalWorkflowConnectionsById(
     proposalWorkflowId: number,
-    proposalWorkflowConnectionId: number
+    proposalWorkflowConnectionId: number,
+    {
+      nextProposalStatusId,
+      prevProposalStatusId,
+    }: {
```
Comment (jekabs-karklins) [inline src/datasources/mockups/ProposalSettingsDataSource.ts:153]: I see you use this type several times. Maybe it makes sense to factor it out into separate file and then reuse the interface

File: src/mutations/InstrumentMutations.ts [hunk 2]
```diff
@@ -267,7 +267,7 @@ export default class InstrumentMutations {
     args: InstrumentSubmitArgs
   ): Promise<InstrumentHasProposals | Rejection> {
     if (
-      !(await this.userAuth.isUserOfficer(agent)) &&
+      !this.userAuth.isUserOfficer(agent) &&
```
Comment (jekabs-karklins) [inline src/mutations/InstrumentMutations.ts:270]: aren't you missing `await` here?

**Review by jekabs-karklins — CHANGES_REQUESTED** [review]
Looks good except the missing `await`

File: src/mutations/InstrumentMutations.ts (hunk 2 above)
Comment (martin-trajanovski) [reply src/mutations/InstrumentMutations.ts:270]: This function isUserOfficer is not an async function. It just checks if that person has that specific role in current role.

**Review by jekabs-karklins — APPROVED** [review]

**Pull request merged**